Add metadata files in ingest

Hwesta opened the pull request:
Added the ability to upload metadata files from a transfer source directory to a SIP in ingest.

From the metadata page in Ingest, added a link to "Add metadata files". Here, you can browse transfer source directories to select files to upload to the SIP's objects/metadata directory. The preferred place to do this is the add metadata wait point, which was moved to before metadata & submission docs are processed.  Converting a metadata.json to a metadata.csv was also added to process metadata files.

Cleaned up the code that parses metadata.csv into file level DC.

metadata.csv files can be added in Transfer and in Ingest. If information conflicts, the Ingest metadata will completely overwrite anything found in transfer. There is also a known bug that if the columns (or column order) differ between the transfer and ingest metadata.csvs, data may end up in the wrong DC tag.

Related to AIP Reingest

Hwesta committed:
Entry filter in browse is configurable

The filter to determine which files/folders are displayed when browsing the
storage service (ie starting a transfer) is now a parameter, instead of
builtin.  Default is still to display folders but not files.

Hwesta committed:
Ingest: Add metadata files structure

Add page to upload new metadata files from.

Hwesta committed:
Ingest: Add metadata files javascript

Add endpoint to copy metadata files. Add JS and templates to display
the browser to select files.

mistydemeo commented on src/MCPClient/lib/clientScripts/archivematicaCreateMETSMetadataCSV.py:
```diff
@@ -51,16 +52,23 @@ def parseMetadata(SIPPath):
+            except Exception:
                 print >>sys.stderr, "error parsing: ", metadataCSVFilePath
-                traceback.print_exc(file=sys.stdout)
+                traceback.print_exc(file=sys.stderr)
```
:thumbsup:

mistydemeo commented:
> metadata.csv files can be added in Transfer and in Ingest. If information conflicts, the Ingest metadata will completely overwrite anything found in transfer.

Let's make sure to log that condition, so there's a debug trail for Archivematica operators who unexpectedly redefine their metadata.

Hwesta commented:
> > metadata.csv files can be added in Transfer and in Ingest. If information conflicts, the Ingest metadata will completely overwrite anything found in transfer.
> 
> Let's make sure to log that condition, so there's a debug trail for Archivematica operators who unexpectedly redefine their metadata.

Not sure that's easily doable currently... I'll look into it.

mistydemeo commented:
Are we planning this for 1.4.0?

Hwesta committed:
METS: Cleanup handling of parsed metadata from CSVs

Clean up data to be sane before passing to helper function. Adjust
helper function parameters and return value to work better with reingest.

Hwesta committed:
Support metadata files added during Ingest

refs #6488

When writing the METS, check for a metadata.csv added to the SIP,
in addition to ones from the originating transfers.  Add conversion
of metadata.json to metadata.csv for files added during ingest.

Move metadata reminder to before processing metadata directory, so new
files are processed.

mistydemeo commented on src/MCPClient/lib/clientScripts/archivematicaCreateMETS2.py:
```diff
@@ -153,18 +155,28 @@ def getDublinCore(unit, id):
 
 
-def createDMDIDSFromCSVParsedMetadataPart2(keys, values):
+def createDmdSecsFromCSVParsedMetadata(metadata):
```
:+1: on new title

mistydemeo commented on src/MCPClient/lib/clientScripts/archivematicaCreateMETSMetadataCSV.py:
```diff
@@ -51,28 +52,36 @@ def parseMetadata(SIPPath):
 
 
-def parseMetadtaCSV(metadataCSVFilePath):
+def parseMetadataCSV(metadataCSVFilePath):
```
Also :+1: on fixing this title

mistydemeo commented:
:+1:
